apps/src/code-studio/components/libraries/LibraryCreationDialog.jsx

```diff
@@ -125,66 +146,94 @@ class LibraryCreationDialog extends React.Component {
   };
 
   displayFunctions = () => {
-    if (!this.state.loadingFinished) {
+    if (this.state.loadingState === LoadingState.LOADING) {
       return (
         <div style={styles.centerSpinner}>
           <Spinner />
         </div>
       );
+    } else if (this.state.loadingState === LoadingState.DONE_LOADING) {
+      let keyIndex = 0;
+      return (
+        <div>
+          <Heading2>
+            <b>{i18n.libraryName()}</b>
+            {this.state.libraryName}
+          </Heading2>
+          <form
+            ref={formElements => {
+              this.formElements = formElements;
+            }}
+            onSubmit={this.publish}
+          >
+            <textarea
+              required
+              name="description"
+              rows="2"
+              cols="200"
+              style={styles.textarea}
+              placeholder="Write a description of your library"
+            />
+            {this.state.sourceFunctionList.map(sourceFunction => {
+              let name = sourceFunction.functionName;
+              let comment = sourceFunction.comment;
+              return (
+                <div key={keyIndex} style={styles.functionItem}>
+                  <input
+                    type="checkbox"
+                    style={styles.largerCheckbox}
+                    disabled={comment.length === 0}
+                    onClick={this.validateInput}
+                    value={keyIndex++}
+                  />
+                  {name}
+                  <br />
+                  {comment.length === 0 && (
+                    <p style={styles.alert}>
+                      {i18n.libraryExportNoCommentError()}
+                    </p>
+                  )}
+                  <pre>{comment}</pre>
+                </div>
+              );
+            })}
+            <input
+              className="btn btn-primary"
+              type="submit"
+              value={i18n.publish()}
+              disabled={!this.state.canPublish}
+            />
+          </form>
+        </div>
+      );
+    } else {
+      return (
+        <div>
+          <Heading2>
+            <b>{i18n.libraryPublishTitle()}</b>
+            {this.state.libraryName}
+          </Heading2>
+          <div>
+            <p style={{fontSize: 20}}>{i18n.libraryPublishExplanation()}</p>
+            <div style={{display: 'flex', justifyContent: 'center'}}>
+              <input
+                type="text"
+                id="library-sharing"
+                onClick={select}
+                readOnly="true"
+                value={dashboard.project.getCurrentId()}
+                style={{cursor: 'copy', width: 300}}
```

Comment:
Can you add `height: 25px` too? I noticed the box and the button are different heights :) 
![image](https://user-images.githubusercontent.com/8324574/67726248-6f5e9b00-f9a2-11e9-9118-9296dbd2dc22.png)